frontend/src/components/project/ProjectDeleteDialog.tsx

```diff
@@ -0,0 +1,172 @@
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { ProjectDefinition } from '../../redux/projectsSlice';
+import { AppDispatch } from '../../redux/stores/store';
+import { DialogTitle } from '../common/Dialog';
+import AuthVisible from '../common/Resource/AuthVisible';
+import { PROJECT_ID_LABEL } from './projectUtils';
+
+interface ProjectDeleteDialogProps {
+  open: boolean;
+  project: ProjectDefinition;
+  onClose: () => void;
+  namespaces: Namespace[];
+}
+
+export function ProjectDeleteDialog({
+  open,
+  project,
+  onClose,
+  namespaces,
+}: ProjectDeleteDialogProps) {
+  const { t } = useTranslation();
+  const dispatch: AppDispatch = useDispatch();
+  const [deleteNamespaces, setDeleteNamespaces] = useState(false);
+
+  const handleDelete = () => {
+    const projectNamespaces = namespaces.filter(ns =>
+      project.namespaces.includes(ns.metadata.name)
+    );
+
+    dispatch(
+      clusterAction(
+        async () => {
+          if (deleteNamespaces) {
+            // Delete the entire namespaces
+            await Promise.all(projectNamespaces.map(namespace => namespace.delete()));
+          } else {
+            // Just remove the project label from each namespace
+            await Promise.all(
+              projectNamespaces.map(async namespace => {
+                const updatedData = { ...namespace.jsonData };
+                if (updatedData.metadata?.labels) {
+                  delete updatedData.metadata.labels[PROJECT_ID_LABEL];
+                  // If labels object becomes empty, remove it entirely
+                  if (Object.keys(updatedData.metadata.labels).length === 0) {
+                    delete updatedData.metadata.labels;
+                  }
+                }
+                return namespace.update(updatedData);
+              })
+            );
+          }
+        },
+        {
+          startMessage: t('Deleting project {{ projectName }}…', { projectName: project.id }),
+          cancelledMessage: t('Cancelled deletion of project {{ projectName }}.', {
+            projectName: project.id,
+          }),
+          successMessage: t('Deleted project {{ projectName }}.', { projectName: project.id }),
+          errorMessage: t('Error deleting project {{ projectName }}.', { projectName: project.id }),
+          successUrl: '/',
+        }
+      )
+    );
+
+    onClose();
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      aria-labelledby="project-delete-dialog-title"
+      aria-describedby="project-delete-dialog-description"
+      maxWidth="sm"
+      fullWidth
+    >
+      <DialogTitle id="project-delete-dialog-title">{t('Delete Project')}</DialogTitle>
+      <DialogContent>
+        <DialogContentText id="project-delete-dialog-description" component="div">
+          <Typography variant="body1" paragraph>
+            {t('What would you like to do with project "{{projectName}}"?', {
```

Comment:
I think this might be more appropriate / in line with our other delete dialogs:
```suggestion
            {t('Are you sure you want to delete project "{{projectName}}"?', {
```